src/herder/TxSetFrame.cpp

```diff
@@ -512,6 +513,8 @@ TxSetFrame::getTotalFees(LedgerHeader const& lh) const
 void
 TxSetFrame::toXDR(TransactionSet& txSet)
 {
+    // Should only call toXDR when in sorted-for-hash state.
+    releaseAssertOrThrow(mHashIsValid);
```

Comment:
I don't think we can guarantee this currently:  some places like `StateSnapshot::writeHistoryBlocks`, or `HerderImpl::persistSCPState` convert to XDR without necessarily calculating the hash (which sets mHashIsValid to true)

Reply:
Mm, ok. I'll call `sortForHash` explicitly. Sigh. This class is .. kinda a mess.